bundle-workflow/python/manifests/build_manifest.py

```diff
@@ -0,0 +1,81 @@
+# Copyright OpenSearch Contributors.
+# SPDX-License-Identifier: Apache-2.0
+
+import yaml
+
+'''
+A BuildManifest is an immutable view of the outputs from a build step
+The manifest contains information about the product that was built (in the `build` section),
+and the components that made up the build in the `components` section.
+
+The format for schema version 1.0 is:
+schema-version: 1.0
+build:
+  name: string
+  version: string
+components:
+  - name: string
+    repository: URL of git repository
+    ref: git ref that was built (sha, branch, or tag)
+    commitid: The actual git commit ID that was built (i.e. the resolved "ref")
```

Comment:
Could we make this `commit_id` or `ref`? Otherwise we'll end up with things like `buildmanifestthatiscool`.

Reply:
`commitid` and `ref` are capturing different things. We only really *need* the `commitid`, but `ref` is helpful for humans to quickly know if this was a build from `main` or `1.0` or whatever branch it was.

I don't quite understand what you mean by `buildmanifestthatiscool`...can you explain?

Reply:
I'd like to be able to feed BuildManifests back into the build system and rebuild components from the same sha.  Use case:  After the initial snapshot build we want to rebuild without a snapshot tag from the same sha's and promote to staging.  If we are using 'ref' from InputManifest, I'd rather use 'ref' and make guarantees that in a BuildManifest this will always be a sha

Reply:
I am just saying that `commitid` should be `commit_id`, underscore, because it's not 1 word.

A "commit ID" is a SHA-1 of all the changes, see https://stackoverflow.com/questions/29106996/what-is-a-git-commit-id. A "ref" is an indirect reference to a commit. The most direct way to reference a commit is via its SHA-1 hash, but it can be a branch, tag, etc.

So I think @mch2 you're saying all the right things and you want a "commit ID", and @camerski is right here to call that a commit ID and not a "ref".

Reply:
For that use case I would generate a new InputManifest from the BuildManifest, rather than reading a BuildManifest and trying to treat it as an InputManifest. Does that make sense?

Reply:
Makes sense - I think a better approach for my use case is to create a separate promotion script that creates a new InputManifest using the commit_id in a built BundleManifest as a component's ref, and then feed it back to the build system.